kedro-datasets/kedro_datasets/spark/spark_stream_dataset.py

```diff
@@ -0,0 +1,158 @@
+            file_format: File format used during load and save
+                operations. These are formats supported by the running
+                SparkContext include parquet, csv, delta. For a list of supported
+                formats please refer to Apache Spark documentation at
+                https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html
+            load_args: Load args passed to Spark DataFrameReader load method.
+                It is dependent on the selected file format. You can find
+                a list of read options for each supported format
+                in Spark DataFrame read documentation:
+                https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html
+            save_args: Save args passed to Spark DataFrame write options.
+                Similar to load_args this is dependent on the selected file
+                format. You can pass ``mode`` and ``partitionBy`` to specify
+                your overwrite mode and partitioning respectively. You can find
+                a list of options for each format in Spark DataFrame
+                write documentation:
+                https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html
+        """
+        self._filepath_ = filepath
+        self.file_format = file_format
+        self._save_args = save_args
+        self._load_args = load_args
+        self.output_format = [
+            "kafka"
+        ]  # message broker formats, such as Kafka, Kinesis, and others, require different methods for loading and saving.
+
+        fs_prefix, filepath = _split_filepath(filepath)
+
+        self._fs_prefix = fs_prefix
+        self._filepath = PurePosixPath(filepath)
+
+        # Handle default load and save arguments
+        self._load_args = deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+    def _describe(self) -> Dict[str, Any]:
+        """Returns a dict that describes attributes of the dataset."""
+        return {
+            "filepath": self._fs_prefix + str(self._filepath),
+            "file_format": self._file_format,
+            "load_args": self._load_args,
+            "save_args": self._save_args,
+        }
+
+    @staticmethod
+    def _get_spark(self):
+        spark_conf_path = "conf/base/spark.yml"
+        if os.path.exists(spark_conf_path):
+            with open(spark_conf_path) as f:
+                self.parameters = yaml.load(f, Loader=SafeLoader)
+            self.spark_conf = SparkConf().setAll(self.parameters.items())
+            spark = SparkSession.builder.config(conf=self.spark_conf).getOrCreate()
+        else:
+            spark = SparkSession.builder.getOrCreate()
+        return spark
```

Comment:
https://docs.kedro.org/en/stable/tools_integration/pyspark.html#initialise-a-sparksession-using-a-hook

In 0.18, this is the preferred way of creating Spark Session and reading spark configs. It's more generic and you can access the config_loader directly.

Reply:
we had the problem with installing Kafka dependency jar file (org.apache.spark:spark-sql-kafka-0-10_2.12:3.3.1) using the generic spark initialisation